scripts/Engineering/gui/engineering_diffraction/tabs/calibration/presenter.py

```diff
@@ -0,0 +1,76 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+# pylint: disable=invalid-name
+from __future__ import (absolute_import, division, print_function)
+
+from mantidqt.utils.asynchronous import AsyncTask
+from mantid.simpleapi import logger
+
+
+class CalibrationPresenter(object):
+    def __init__(self, model, view):
+        self.model = model
+        self.view = view
+        self.view.set_on_calibrate_clicked(self.on_calibrate_clicked)
+        self.worker = None
+        self.instrument = "ENGINX"
+        self.rb_num = None
+
+    def on_calibrate_clicked(self):
+        if not self.validate_run_numbers():
+            return
+        if self.view.is_searching():
+            return
+        vanadium_no = self.view.get_vanadium_filename()
+        calib_no = self.view.get_calib_filename()
+        plot_output = self.view.get_plot_output()
+        self.start_calibration_worker(vanadium_no, calib_no, plot_output, self.rb_num)
+
+    def start_calibration_worker(self, vanadium_path, calib_path, plot_output, rb_num):
+        """
+        Calibrate the data in a separate thread so as to not freeze the GUI.
+        :param vanadium_path: Path to vanadium data file.
+        :param calib_path: Path to calibration data file.
+        :param plot_output: Whether to plot the output.
+        :param rb_num: The current RB number set in the GUI.
+        """
+        self.worker = AsyncTask(self.model.create_new_calibration, (vanadium_path, calib_path),
+                                {"plot_output": plot_output, "instrument": self.instrument, "rb_num": rb_num},
+                                error_cb=self._on_error, finished_cb=self.enable_calibrate_buttons)
+        self.worker.start()
+        self.disable_calibrate_buttons()
```

Comment:
Maybe we should disable the buttons first and then start the processing? That way if we reduce the amount of time it's possible to click the button twice.